server/game/gameSystems/PayResourceCostSystem.ts

```diff
@@ -37,7 +37,7 @@ export class PayResourceCostSystem<TContext extends AbilityContext = AbilityCont
 
     public override canAffect(player: Player, context: TContext, additionalProperties = {}): boolean {
         const properties = this.generatePropertiesFromContext(context, additionalProperties);
-        return properties.amount > 0 && player.countSpendableResources() > 0 && super.canAffect(player, context, additionalProperties);
+        return properties.amount > 0 && player.countSpendableResources() > 0 && super.canAffect(player, context, additionalProperties) && player.countSpendableResources() >= properties.amount;
```

Comment:
Good catch! Can you also do two small bookkeeping favors for me:

1. At the top of this class, put the comment `//TODO: replace this with ExhaustResourcesSystem when we have it`
2. Add an override property named `defaultProperties` that sets `amount: null` and `isCost: true`. You can look at e.g. GiveShieldSystem for an example.